.github/scripts/prepare-registry-constructor.js

```diff
@@ -0,0 +1,145 @@
+ * @param {string} [options.descriptorPath] - Path to existing registry descriptor (required if registryExists is true)
+ * @returns {Object} constructor - The constructor object that is created from the template.
+ */
+export function prepareRegistryConstructor(options) {
+    const {
+        constructorPath,
+        registryVersion,
+        pluginName,
+        pluginComponent,
+        pluginVersion,
+        registryExists,
+        descriptorPath
+    } = options;
+
+    const template = fs.readFileSync(constructorPath, 'utf8');
+    const constructor = yaml.load(template);
+    constructor.version = registryVersion;
+
+    if (registryExists) {
+        const descriptor = JSON.parse(fs.readFileSync(descriptorPath, 'utf8'));
+        constructor.componentReferences = descriptor.componentReferences || [];
+
+        if (constructor.componentReferences.find(r => {
+            return r.name === pluginName && r.version === pluginVersion;
+        })) {
+            throw new Error(`Plugin with name ${pluginName} and version ${pluginVersion} already exists in reference list`);
+        }
+
+        // Compute the new version. If the plugin does not exist we increase the minor version.
+        const pluginExists = constructor.componentReferences.find(r => r.name === pluginName);
+        const nextVersion = computeNextVersions(registryVersion, registryVersion, "", pluginExists);
+        constructor.version = nextVersion.baseVersion;
+
+    } else {
+        if (!constructor.componentReferences) {
+            constructor.componentReferences = [];
+        }
+
+        constructor.version = "v0.0.1"
+    }
+
+    const plugin = {
+        name: pluginName,
+        componentName: pluginComponent,
+        version: pluginVersion
+    };
+    constructor.componentReferences.push(plugin)
+
+    return {constructor}
+}
+
+/**
+ * GitHub Actions entrypoint for preparing registry constructor.
+ *
+ * Environment variables:
+ * - REGISTRY_CONSTRUCTOR: Path to constructor template file (required)
+ * - REGISTRY_VERSION: New registry version (required)
+ * - PLUGIN_NAME: Plugin name (required)
+ * - PLUGIN_COMPONENT: Plugin component name (required)
+ * - PLUGIN_VERSION: Plugin version (required)
+ * - REGISTRY_EXISTS: Whether registry exists - "true" or "false" (required)
+ * - DESCRIPTOR_PATH: Path to existing registry descriptor (required if REGISTRY_EXISTS is "true")
+ *
+ * @param {import('@actions/github-script').AsyncFunctionArguments} args
+ */
+export default async function prepareRegistryConstructorAction({ core }) {
+    const constructorPath = process.env.REGISTRY_CONSTRUCTOR;
+    const registryVersion = process.env.REGISTRY_VERSION;
+    const pluginName = process.env.PLUGIN_NAME;
+    const pluginComponent = process.env.PLUGIN_COMPONENT;
+    const pluginVersion = process.env.PLUGIN_VERSION;
+    const registryExists = process.env.REGISTRY_EXISTS === 'true';
+    const descriptorPath = process.env.DESCRIPTOR_PATH;
```

Comment:
why are we passing so much craziness into this script instead of just running the cli from within the script? feels a lot easier than the many steps in the github action

Reply:
@jakobmoellerdev I endeavoured on this quest a bit. I can shift the OCM calls into the Javascript and remove two more steps from the YAML GitHub Action. Would that be better?